### PR TITLE
Fix externally controlled executor not pausing

### PR DESCRIPTION
**Pull request by MStoykov**

### CODE REVIEW

File: lib/executor/vu_handle.go [hunk 1]
```diff
@@ -41,7 +42,11 @@ type vuHandle struct {
 	returnVU  func(lib.InitializedVU)
 	config    *BaseConfig
 
+	vu           lib.InitializedVU
 	canStartIter chan struct{}
+	// This is here only to signal that something has changed it must be added to and read with atomics
+	// and helps to skip checking all the contexts and channels all the time
```
Comment (imiric) [inline lib/executor/vu_handle.go:48]: Maybe it helps to clarify:

```suggestion
	// If change is not 0, it signals that the VU needs to be reinitialized. It must be added to and read
	// with atomics and helps to skip checking all the contexts and channels all the time.
```

File: lib/executor/variable_looping_vus.go [hunk 2]
```diff
@@ -614,7 +614,7 @@ func (vlv VariableLoopingVUs) Run(ctx context.Context, out chan<- stats.SampleCo
 	handleNewScheduledVUs := func(newScheduledVUs uint64) {
 		if newScheduledVUs > currentScheduledVUs {
 			for vuNum := currentScheduledVUs; vuNum < newScheduledVUs; vuNum++ {
-				vuHandles[vuNum].start()
+				_ = vuHandles[vuNum].start() // TODO handle error
```
Comment (imiric) [inline lib/executor/variable_looping_vus.go:617]: Would this involve a larger VLV refactor? Are we sure we can postpone it?

Comment (MStoykov) [reply lib/executor/variable_looping_vus.go:617]: we've been ignoring the errors up until now ... so ... I would say - yes, although if possible we should probably prioritise it

File: lib/executor/vu_handle.go [hunk 3]
```diff
@@ -102,63 +164,68 @@ func (vh *vuHandle) hardStop() {
...
 
-		// Wait for either the executor to be done, or for us to be un-paused
 		select {
-		case <-canStartIter:
-			// Best case, we're currently running, so we do nothing here, we
-			// just continue straight ahead.
 		case <-executorDone:
 			// The whole executor is done, nothing more to do.
 			return
 		default:
```
[the remark this answers is not in the document]
Comment (MStoykov) [reply lib/executor/vu_handle.go:197]: yeah

File: lib/executor/vu_handle.go (hunk 1 above)
Comment (MStoykov) [reply lib/executor/vu_handle.go]: resolved in 02e96f6

File: lib/executor/vu_handle.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (MStoykov) [reply lib/executor/vu_handle.go:197]: resolved with d314f3a

**Pull request merged**